src/stream/src/executor/dynamic_filter.rs

```diff
@@ -60,6 +60,11 @@ pub struct DynamicFilterExecutor<S: StateStore, const USE_WATERMARK_CACHE: bool>
     metrics: Arc<StreamingMetrics>,
     /// The maximum size of the chunk produced by executor at a time.
     chunk_size: usize,
+    /// If the right side's change always make the condition more relaxed.
+    /// In other words, make more record in the left side satisfy the condition.
+    /// In that case, there are only records which does not satisfy the condition in the table.
```

Comment:
IIUC, this means we store records which do not satisfy the condition,

so if the condition is something like:
```
LHS < NOW()
```

We only store
```
LHS >= NOW()
```

But why do we store records which do not satisfy the condition in the table?

Reply:
Oh I got it.
Consider the following case with the following records in upstream:
```
1
2
3
4
```

And the current filter (RHS < NOW()) is:
```
3
```

We store internal state, which only stores values which are excluded:
```
3
4
```

When RHS increases to 5,
```
3
4
```

We can now emit 3, 4 to downstream.